src/plugins/vis_type_timeseries/public/application/components/annotations_editor.js

```diff
@@ -83,6 +96,10 @@ export class AnnotationsEditor extends Component {
       const fn = collectionActions.handleChange.bind(null, this.props);
       fn(_.assign({}, model, part));
     };
+    const changeFetchedIndex = async (index) => {
+      handleChange(index);
+      await this.setFetchedIndex(index[INDEX_PATTERN_NAME]);
```

Comment:
how it should work in case of 2 or more annotations with different index patterns? Looks like you set in state the last `fetchedIndex` which will be used for each annotations.